src/fsharp/DotNetFrameworkDependencies.fs

```diff
@@ -34,6 +34,98 @@ module internal FSharp.Compiler.DotNetFrameworkDependencies
+    //
+    // Format:
+    // =======
+    //      $(Major).$(Minor).$(Build) [-SomePrefix]
+    //   Major, Minor, Build collates normally
+    //   Strings without -SomePrefix collate higher than SomePrefix, 
+    //   SomePrefix collates using normal alphanumeric rules
+    //
+    let deconstructVersion (version:string)  =
+        let getSuffix =
+            let pos = version.IndexOf("-")
+            if pos >= 0 then version.Substring(pos + 1) else ""
+        let elements = version.Split('.')
+
+        if elements.Length < 3 then
+            struct (0, 0, 0, getSuffix)
+        else
+            struct (Int32.Parse(elements.[0]), Int32.Parse(elements.[1]), Int32.Parse(elements.[2]), getSuffix)
+
+    let versionCompare c1 c2 =
+        if c1 = c2 then 0
+        else
+            try
+                let struct (major1, minor1, build1, suffix1 ) = deconstructVersion c1
+                let struct (major2, minor2, build2, suffix2 ) = deconstructVersion c2
+
+                let v = major2 - major1
+                if v = 0 then 0
+                else
+                    let v = minor2 - minor1
+                    if v <> 0 then v
+                    else
+                        let v = build2 - build1
+                        if v <> 0 then v
+                        else
+                            String.CompareOrdinal(suffix2, suffix1)
+            with _ -> 0
+
+    let executionTfm =
+        let file =
+            try
+                let depsJsonPath = Path.ChangeExtension(Assembly.GetEntryAssembly().Location, "deps.json")
+                File.ReadAllText(depsJsonPath)
+            with _ -> ""
+
+        let tfmPrefix=".NETCoreApp,Version=v"
+        let pattern = "\"name\": \"" + tfmPrefix
+        let startPos =
+            let startPos = file.IndexOf(pattern, StringComparison.OrdinalIgnoreCase)
+            if startPos >= 0  then startPos + (pattern.Length) else startPos
+
+        let length =
+            if startPos >= 0 then
+                let ep = file.IndexOf("\"", startPos)
+                if ep >= 0 then ep - startPos else ep
+            else -1
+        match startPos, length with
+        | -1, _
+        | _, -1 -> None
+        | pos, length -> Some ("netcoreapp" + file.Substring(pos, length))
+
+
+    let getFrameworkRefsPackDirectoryPath =
+        match executionTfm with
+        | Some _ ->
+            let appRefDir = Path.Combine(getFSharpCompilerLocation, "../../../packs/Microsoft.NETCore.App.Ref")
+            if Directory.Exists(appRefDir) then
+                Some appRefDir
+            else
+                None
+        | _ -> None
+
+    let isInReferenceAssemblyPackDirectory filename =
+        match getFrameworkRefsPackDirectoryPath with
+        | Some appRefDir ->
+            let path = Path.GetDirectoryName(filename)
+            path.StartsWith(appRefDir, StringComparison.OrdinalIgnoreCase)
+        | _ -> false
+
+    let getFrameworkRefsPackDirectory =
```

Comment:
This looks like it should be a function rather than a computation.  Also have `///` comments on all of these would be good please